image-capture/update-camera-zoom.js

```diff
@@ -3,24 +3,24 @@ navigator.mediaDevices.getUserMedia({video: true})
   document.querySelector('video').srcObject = mediaStream;
 
   const track = mediaStream.getVideoTracks()[0];
-  const imageCapture = new ImageCapture(track);
+  const capabilities = track.getCapabilities();
+  const settings = track.getSettings();
 
-  return imageCapture.getPhotoCapabilities()
-  .then(photoCapabilities => {
-    // Check whether zoom is supported or not.
-    if (!photoCapabilities.zoom.min && !photoCapabilities.zoom.max) {
-      return Promise.reject('Zoom is not supported by ' + track.label);
-    }
+  const input = document.querySelector('input[type="range"]');
 
-    // Map zoom to a slider element.
-    const input = document.querySelector('input[type="range"]');
-    input.min = photoCapabilities.zoom.min;
-    input.max = photoCapabilities.zoom.max;
-    input.step = photoCapabilities.zoom.step;
-    input.value = photoCapabilities.zoom.current;
-    input.oninput = function(event) {
-      imageCapture.setOptions({zoom: event.target.value});
-    }
-  });
+  // Check whether zoom is supported or not.
+  if (!capabilities.zoom.min && !capabilities.zoom.max) {
+    return Promise.reject('Zoom is not supported by ' + track.label);
+  }
+
+  // Map zoom to a slider element.
+  input.min = capabilities.zoom.min;
+  input.max = capabilities.zoom.max;
+  input.step = capabilities.zoom.step;
+  input.value = settings.zoom;
+  input.oninput = function(event) {
+    track.applyConstraints({zoom: event.target.value});
```

Comment:
@miguelao Please update example from https://www.w3.org/TR/image-capture/#update-camera-zoom-and-code-takephoto-code- then.
Thanks!